src/game/objects/general/bridge.c

```diff
@@ -73,20 +76,51 @@ static int32_t Bridge_GetOffset(
     return offset;
 }
 
+static void Bridge_FixEmbedded(int16_t item_num)
+{
+    ITEM_INFO *item;
+    FLOOR_INFO *floor;
+    int32_t x, y, z;
+    int16_t room_number, floor_height, old_anim, old_frame, bridge_height;
+    int16_t *bounds;
+
+    item = &g_Items[item_num];
+    if (item->status != IS_ACTIVE) {
+        x = item->pos.x;
+        y = item->pos.y;
+        z = item->pos.z;
+        room_number = item->room_number;
+
+        floor = Room_GetFloor(x, y, z, &room_number);
+        floor_height = Room_GetHeight(floor, x, y, z);
+
+        // Only move the bridge up if it's at floor level.
+        if (item->floor == floor_height) {
+            bounds = Item_GetBoundsAccurate(item);
+            bridge_height =
+                ABS(bounds[FRAME_BOUND_MAX_Y]) - ABS(bounds[FRAME_BOUND_MIN_Y]);
+            item->pos.y = floor_height - bridge_height;
+        }
+    }
+}
+
 void Bridge_SetupFlat(OBJECT_INFO *obj)
 {
+    obj->initialise = Bridge_FixEmbedded;
```

Comment:
Let's be consistent with `bat.c`:

1. `obj->initialise = Bridge_Initialise`
2. ```c
    void Bridge_Initialise(int16_t item_num)
    {
        // Some comment about embedded bridges
        Bridge_FixEmbeddedPosition(item_num);
    }
    ```

Note how initialise is not static. It goes in .h too.